Add netifd script to control SSTP connections.

rkkoszewski opened the pull request:
Description: Add netifd script to control SSTP connections.

Script was made by harveyhu2012 and slightly modified by me (rkkoszewski)

It is required for luci-proto-sstp (https://github.com/openwrt/luci/pull/2768)

rkkoszewski commented on net/sstp-client/files/lib/netifd/proto/sstp.sh:
```diff
@@ -0,0 +1,130 @@
+		
+	# WORKAROUND: Workaround to properly register the sstp interface (As seeen in: https://forum.archive.openwrt.org/viewtopic.php?id=58007)
+	# WORKAROUND: Start
+	sleep 10
```
Anybody has a better idea as an approach for this? Setting a fixed timeout seems hacky at best. From what I understand, the reason this fails without the timeout is that the sstpc starts a pppd subprocess asyncronously and if the next commands get invoked before it started, they fail to register the new device. 

I don't know if it's somehow possible to get the PID of the currently running proto_run_command, if yes, then pgrep could be used to check for the subprocess every second or so, or if the proto_run_command process dies then just abort. But so far I was unable to find the PID of the proto_run_command programatically. Anybody has an idea?

rkkoszewski commented:
Anybody can suggest what to do regarding the missing signoff? The original author @harveyhu2012 doesn't seem to be active on GitHub since 2014.

neheb commented:
Needs a PKG_RELEASE bump. Also, please rebase against master as there's a random merge commit in there.

rkkoszewski commented:
> Needs a PKG_RELEASE bump. Also, please rebase against master as there's a random merge commit in there.

Rebased against master and bumped the package version. Haven't yet worked with the rebase command and I added the workaround from my commit to the first one by mistake, therefore my workaround commit appears as just a space (But it's signed off by me). Squashing should fix that issue though.

neheb commented:
Maintainer is @fededim

rkkoszewski commented on net/sstp-client/Makefile:
```diff
@@ -8,7 +8,7 @@
 
 PKG_NAME:=sstp-client
-PKG_VERSION:=1.0.12
+PKG_VERSION:=1.1.0
```
Hi @fededim, sorry but I mistook PKG_RELEASE with PKG_VERSION. I will correct it in the next push.

rkkoszewski replied:
Ok, so this got fixed in this commit now: https://github.com/openwrt/packages/pull/9221/commits/045149cb217452d04c13bf6d1afdfcc388fc52b6

fededim replied:
Ok changes are fine for me

fededim replied:
Remember to squash and amend your commits, otherwise you can't merge

rkkoszewski commented:
Is there still something missing in this PR preventing it from being merged? @neheb @fededim

neheb commented:
Maintainer approved. DCO is failing because of mismatching author name. Will merge anyway.

BKPepe commented:
@neheb I think we have those checks for a reason. The DCO can be skipped, so CircleCI can try to build it. Never merge anything when there are 0 of 2 checks passed. 

Also, commits should be squashed as maintainer requested.

rkkoszewski commented:
Hi @BKPepe, sorry I thought that GitHub does the squashing automatically (Was my first contribution here, although it seems like it did squash it from what I can see in the commits). Is there any advantage of performing it manually?
